### PR TITLE
Qualcomm AI Engine Direct -- update documents

### PR DESCRIPTION
**Pull request by chiwwang**
1. Updated QNN download links.
2. Unified setup.md and build-run-qualcomm-ai-engine-direct-backend.md
3. Fixed build commands in build-run-qualcomm-ai-engine-direct-backend.md

### CODE REVIEW

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md [hunk 1]
```diff
@@ -107,22 +109,38 @@ export PYTHONPATH=$EXECUTORCH_ROOT/..
...
 
 ### AOT (Ahead-of-time) components:
 
 Python APIs on x64 are required to compile models to Qualcomm AI Engine Direct binary.
 
 ```bash
 cd $EXECUTORCH_ROOT
-# Workaround for fbs files in exir/_serialize
-cp schema/program.fbs exir/_serialize/program.fbs
-cp schema/scalar_type.fbs exir/_serialize/scalar_type.fbs
```
Comment (cccclai) [inline docs/source/build-run-qualcomm-ai-engine-direct-backend.md:120]: Are we good with this now?

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md [hunk 2]
```diff
@@ -107,22 +109,38 @@ export PYTHONPATH=$EXECUTORCH_ROOT/..
...
-
 mkdir build_x86_64
 cd build_x86_64
```
Comment (cccclai) [inline docs/source/build-run-qualcomm-ai-engine-direct-backend.md:130]: let's add them to git ignore?

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md [hunk 3]
```diff
@@ -107,22 +109,38 @@ export PYTHONPATH=$EXECUTORCH_ROOT/..
...
+  -DCMAKE_INSTALL_PREFIX=$PWD \
+  -DEXECUTORCH_BUILD_QNN=ON \
+  -DQNN_SDK_ROOT=${QNN_SDK_ROOT} \
+  -DEXECUTORCH_BUILD_SDK=ON \
+  -DEXECUTORCH_BUILD_EXTENSION_MODULE=ON \
+  -DEXECUTORCH_ENABLE_EVENT_TRACER=ON \
+  -DPYTHON_EXECUTABLE=python3 \
+  -DEXECUTORCH_SEPARATE_FLATCC_HOST_PROJECT=OFF
+
+cmake --build $PWD -t "PyQnnManagerAdaptor" "PyQnnWrapperAdaptor" -j8
```
Comment (cccclai) [inline docs/source/build-run-qualcomm-ai-engine-direct-backend.md:143]: Any specific reason we use `-j8` here and `-j16` in the following cmake command?

**Review by cccclai — APPROVED** [review]
Thanks for updating the documents! Just some minor comments. Should we add a session for x86 simulator? I find it super helpful. And possibly a model supported list.

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md (hunk 1 above)
Comment (chiwwang) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:120]: No. But this action is more related to AOT compilation, so I moved there:
https://github.com/pytorch/executorch/pull/4590/files#diff-8bac58bd3ddef654a20549948dfea5d27f76a8ee29544328f65819cc0ccd7bcfR205

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md (hunk 2 above)
Comment (chiwwang) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:130]: I'm fine with that... but how about using the name already in `.gitignore`?
```
cmake-out/
cmake-android-out/
cmake-out-android/
```
I can change `build_x86_64` to `cmake-out` and `build_android` to `cmake-android-out`. It seems more consistent with other backends.

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md (hunk 3 above)
Comment (chiwwang) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:143]: No. I will fix this to avoid confusion.

**Comment by chiwwang** [comment]
> Thanks for updating the documents! Just some minor comments. Should we add a session for x86 simulator? I find it super helpful. And possibly a model supported list.

Thank you for the promot review!
I will do this in days and notify you.

**Comment by chiwwang** [comment]
Turning to draft before
1. address feedback
2. add x86 emulator section
3. mention support model list (basically, those in `examples/qualcomm`)

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md (hunk 2 above)
Comment (cccclai) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:130]: That sounds good

**Comment by chiwwang** [comment]
Hi @cccclai ,
I addressed the comments. Please take a look if you happen to have some time :)

**Review by cccclai — APPROVED** [review]

**Comment by guangy10** [comment]
I ran into issues with the updated guide:
`./backends/qualcomm/scripts/build.sh --release`
```
/data/users/guangyang/executorch/examples/qualcomm/llama2/qaihub_runner/runner.cpp:21:10: fatal error: 'executorch/examples/models/llama2/runner/util.h' file not found
   21 | #include <executorch/examples/models/llama2/runner/util.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by cccclai** [comment]
Looks like cause by https://github.com/pytorch/executorch/pull/4588

**Pull request merged**